src/main/java/org/testng/xml/internal/XmlSuiteUtils.java

```diff
@@ -7,47 +7,40 @@
 import org.testng.xml.XmlSuite;
 import org.testng.xml.XmlTest;
 
+import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
 
 /**
  * A utility class that exposes helper methods to work with {@link XmlSuite}
  */
 public final class XmlSuiteUtils {
-
+    private static List<XmlSuite> cloneSuites = Lists.newArrayList();
```

Comment:
I am not very comfortable with managing states via static variables and then explicitly relying on reset methods to reset states as well. Can we have this refactored such that we dont have to rely on those methods ?

If we really would want to maintain states, then we should make `XmlSuiteUtils` as an instance class, and we instantiate it and then work with the created instance.

Reply:
We share the same view on it. If you are fine with creating instance instead of Singleton, I am happy to change it.

Reply:
@eyehwan - Once again my apologies if my comment misled you into believing that I was hinting at singletons. No I was **not referring to singletons** when I said can we work with an instance.
I was referring to us trying to work with a regular instance of `XmlSuiteUtils` which would store the state and its instance gets passed around to all methods that need the state to work.

Singletons, make writing unit tests even more harder.